# Homework #2

## 1. Solid or Gas Earth?

In class, we defined the quantity:

$$\Gamma = \frac{E_\mathrm{Coul}}{kT}$$

as a measure of the importance of the Coulomb energy to the thermal energy.  If $\Gamma \gg 1$, then we are a solid.  We got the separation between nuclei from the number density and the temperature from the Virial theorem and got:

$$\Gamma = \frac{Z^2 e^2}{A^{4/3} m_u^{4/3} G M^{2/3}}$$

Now for the earth, we can take $A$ and $Z$ to just be for iron

In [1]:
A = 56
Z = 26

For the constants, we'll use a library, `unyt` to get them along with the units

In [2]:
import unyt

In [3]:
e = unyt.qe.in_cgs()
m_u = (1.0 * unyt.amu).in_cgs()
G = unyt.G.in_cgs()
M = unyt.Mearth.in_cgs()

Notice that these all have units

In [4]:
m_u

unyt_quantity(1.66053892e-24, 'g')

In [5]:
e

unyt_quantity(-4.80320467e-10, 'statC')

In [6]:
gamma = (Z * e)**2 / ((A * m_u)**(4./3.) * G * M**(2./3.))
gamma

unyt_quantity(167.13713567, '(dimensionless)')

So this tells us that gamma is dimensionless!

$$\Gamma = 167$$

Now, really that we need to be careful with Coulomb's law and units.  In CGS, the energy is:

$$E_\mathrm{Coul} = \frac{q_1 q_2}{r}$$

but in SI, it is:

$$E_\mathrm{Coul} = \frac{1}{4\pi \epsilon_0} \frac{q_1 q_2}{r}$$

Let's redo that calculation but using SI units:

In [7]:
e = unyt.qe
m_u = 1 *unyt.amu
G = unyt.G
M = unyt.Mearth

In [8]:
gamma = (Z * e)**2 / ((A * m_u)**(4./3.) * G * M**(2./3.))
gamma

unyt_quantity(2.20215509e-17, 'C**2*s**2/(amu**(1/3)*kg**(2/3)*m**3)')

Notice that the units don't work now!  Let's add in the $1/(4\pi \epsilon_0)$

In [9]:
import math

In [10]:
eps0 = unyt.eps_0

In [11]:
gamma = (Z * e)**2 / ((A * m_u)**(4./3.) * G * M**(2./3.)) / (4 * math.pi * eps0)
gamma

unyt_quantity(167.13713567, '(dimensionless)')

## 2. Ideal gas

In [ ]:
from sympy import init_session
init_session(use_latex="mathjax")

In [ ]:
ni, m, k, T = symbols("n_I m k_B T", real=True, positive=True)
p, v = symbols("p v", real=True)
n_I = symbols("n_I", real=True)
g = symbols("g", real=True, positive=True)
h = symbols("h", real=True, positive=True)
E_0 = symbols("E_0", real=True)
mu = symbols("mu", real=True)

### a. Finding Maxwell

We start with

In [ ]:
n = 4 * pi * p**2 * g / h**3 * exp(mu/(k*T)) * exp(-E_0/(k*T)) * exp(-p**2/(2*m*k*T))
n

now let's integrate over p

In [ ]:
n_I_int = integrate(n, (p, 0, oo))
n_I_int

This is equal to $n_I$.  We want to substitute this back into $n(p)$.  Let's solve for $g$ in terms of $n_I$
and then use that as a way to substitute this back.

In [ ]:
g_solve = solve(Eq(n_I_int, n_I), g)[0]

In [ ]:
g_solve

In [ ]:
n = simplify(n.subs(g, g_solve, strict=False))
n

Now we see that we have the normal Maxwell-Boltzmann distribution.

### b. Ideal gas pressure

The pressure integral is simply:

$$P = \frac{1}{3} \int_0^\infty n(p) v p dp$$

with $v = p / m$.

In [ ]:
P = integrate(Rational(1, 3) * n * p**2 / m, (p, 0, oo))
P

We see that we get the familiar ideal gas law!

## c. Energy

The energy is:

$$\rho e = \int_0^\infty n(p) E(p) dp$$

with $E(p) = p^2 / (2m)$

In [ ]:
rhoe = integrate(n * p**2 / (2 *m), (p, 0, oo))
rhoe

So we see that this is $3/2 P$.